week3/Q1.py

```diff
@@ -0,0 +1,177 @@
+				vertices.append({i + 1, i - n, i - n + 1})
+			#down sight
+			elif i > n*(m - 1):
+				vertices.append({i - 1, i + 1, i - n, i - n - 1, i - n + 1})
+			#up sight
+			elif i < n:
+				vertices.append({i - 1, i + 1, i + n, i + n - 1, i + n + 1})
+			#left sight
+			elif i % n == 0:
+				vertices.append({i + 1, i - n, i + n, i - n + 1, i + n + 1})
+			#right sight
+			elif (i + 1) % n == 0:
+				vertices.append({i - 1, i - n, i + n, i - n - 1, i + n - 1})
+			#center
+			else:
+				vertices.append({i - 1, i + 1, i - n, i + n, i - n - 1, i - n + 1, i + n - 1, i + n + 1})
+
+	return {'nodes': nodes, 'vertices': vertices}
+
+
+def _get_prefixes_list(dictionary):
+	'''
+	Gets list of all possible prefixes in dictionary.
+
+	Args:
+		dictionary: set
+
+	Returns:
+		set, all possible prefixes
+	'''
+	prefixes = set()
+	for word in dictionary:
+		for i, letter in enumerate(word):
+			prefixes.add(word[:i + 1])
+	return prefixes
+
+
+def _isWord(line, dictionary):
+	'''
+	Checks if sequence of elements is a real word.
+
+	Args:
+		line: str, sequence of elements
+		dictionary: set
+
+	Returns:
+		boolean
+	'''
+	if line in dictionary:
+		return True
+	else:
+		return False
+
+
+def _isPrefix(line, prefixes):
+	'''
+	Checks if sequence of elements is a possible prefix.
+
+	Args:
+		line: str, sequence of elements
+		prefixes: set
+
+	Returns:
+		boolean
+	'''
+	if line in prefixes:
+		return True
+	else:
+		return False
+
+
+def _walk_table(line, node, table, dictionary, prefixes, visited_nodes):
+	'''
+	Walks through the table, looking for words.
+
+	Args:
+		line: str, sequence of elements from visited nodes
+		node: int, number of the last visited node
+		table: dict
+		dictionary: set
+		prefixes: set
+		visited_nodes: set of int, nodes that had been visited and can't be used again
+
+	Returns:
+		set, words from dictionary that can be build from the chosen node
+	'''
+	words = set()
+	if _isPrefix(line, prefixes):
+		if _isWord(line, dictionary):
+			words.add(line)
+		for el in table['vertices'][node]:
+			if el not in visited_nodes:
+				new_line = line + table['nodes'][el]
+				words.update(_walk_table(new_line, el, table, dictionary, prefixes, visited_nodes | {el}))
+	return words
+
+
+
+def get_words_in_table(dictionary, table):
+	'''
+	Gets all real words from the table.
+
+	Args:
+		dictionary: set
+		table: set
+
+	Returns:
+		set, all real words in the table 
+	'''
+	words = set()
+	prefixes = _get_prefixes_list(dictionary)
+	for node in table['nodes']:
+		words.update(_walk_table(table['nodes'][node], node, table, dictionary, prefixes, {node}))
+	return words
+
+
+def main():
+	n = int(input('Input number of columns: '))
+	m = int(input('Input number of rows: '))
+	print('Input table row by row:')
+	inp = input()
+	values = ''
+	while inp:
+		values += inp
+		inp = input()
+	print('Input dictionary word by word:')
+	inp = input()
+	dictionary = set()
+	while inp:
+		dictionary.add(inp)
+		inp = input()
+	table = get_graph(values, n, m)
```

Comment:
Since your `main` function creates the graph, you cannot test. Given that the problem was written as having the grid as an input, you could put these two lines together in a function, and be able to test it all together.